### PR TITLE
fix(interactions): deliver question answers durably

### PR DESCRIPTION
**Pull request by cryppadotta**
## Thinking Path

> - Paperclip is the open source app people use to manage AI agents for work.
> - Agents can pause a task and ask the user structured questions.
> - The answer is durable in the issue interaction, but delivery to the next run is not durable.
> - A process restart can therefore leave an answered interaction without a continuation attempt.
> - Native runners also need a provider-neutral question contract before the task page can consume native events safely.
> - This pull request adds a content-free delivery outbox and an optional native steering seam.
> - Direct adapters keep their existing heartbeat continuation path.
> - The benefit is reliable answer delivery without changing runtime selection or task-page behavior.

## Linked Issues or Issue Description

Refs #12202. This pull request replaces the question-delivery foundation from that stale task-thread pull request. The task-thread projection will follow in a smaller pull request.

**What happened?**

Question answers were stored in the issue interaction. The server then made one in-memory continuation wake. A server stop between those operations could leave the answer stored but not delivered. The combined native task-thread pull request also made this behavior hard to review separately from UI changes.

**Expected behavior**

The answer and its delivery receipt must commit in one transaction. The server must retry pending receipts after a restart. Existing direct adapters must keep the current wake path. A native runtime may use the optional steering seam, but this pull request does not enable native steering in production.

**Steps to reproduce**

1. Create an `ask_user_questions` interaction.
2. Answer the interaction.
3. Stop the server before the continuation wake completes.
4. Start the server again.
5. On current master, no durable record tells the server to retry the answer delivery.

**Paperclip version or commit**

Current `master` at `4d82f5eae`.

## What Changed

- Add the `issue_question_response_deliveries` table and migration.
- Store only routing state, a correlation ID, and a payload digest in the delivery row. The answer remains in the existing interaction result.
- Commit an answered interaction and its pending delivery row in one transaction.
- Add bounded claims, retry recovery, cumulative terminal state, and content-free activity records.
- Keep every built-in direct adapter and external adapter on the existing heartbeat wake path.
- Add an optional native steering seam. No production caller supplies that seam in this pull request.
- Retain the provider-neutral `paperclip.question_set.v1` presentation on recovered interactions.
- Run delivery immediately after an answer and sweep pending rows at startup and on the existing server interval.
- Add focused database, service, route, startup, adapter-matrix, digest, and duplicate-delivery tests.

## Compatibility Boundary

- This pull request does not change adapter selection.
- This pull request does not start runnerd.
- This pull request does not create native run records.
- Direct adapters never call the native steering seam.
- The existing interaction result stays authoritative for answer content.
- The migration is additive and does not rewrite existing rows.
- This pull request has no UI, dependency, workflow, package-manager, or lockfile changes.
- The diff has 19 files.

## Verification

- `pnpm exec vitest run server/src/__tests__/question-response-delivery.test.ts server/src/services/issue-thread-interactions.test.ts server/src/__tests__/issue-thread-interaction-routes.test.ts server/src/__tests__/server-startup-feedback-export.test.ts` — 4 files and 120 tests passed.
- `pnpm -r typecheck` — passed for all applicable workspaces. This includes Cargo format and check, protocol drift checks, and migration safety.
- `pnpm build` — passed. This includes the Rust release binary, server build, and UI production build.
- `git diff --check` — passed.
- Secret patterns were not present in the changed text files.
- The repository token gates currently report violations from unchanged files on `master`. This pull request does not change those files.

## Risks

The main risk is routing a direct-adapter answer into a native session. The service checks the persisted runtime mode, and the adapter matrix proves that all direct adapters use only the existing wake path. The new table is additive. It has foreign keys, unique correlation constraints, bounded attempts, and status checks. Activity records omit question and answer content.

## Model Used

OpenAI Codex, GPT-5 family. The client does not expose the exact deployment ID or context window. Agentic reasoning, tool use, and code execution were enabled.

## Checklist

- [x] I have included a thinking path that traces from project context to this change
- [x] I have specified the model used (with version and capability details)
- [x] I have checked ROADMAP.md and confirmed this PR does not duplicate planned core work
- [x] I have searched GitHub for duplicate or related PRs and linked them above
- [x] I have either (a) linked existing issues with `Fixes:` / `Closes:` / `Refs:` OR (b) described the issue in-PR following the relevant issue template
- [x] I have not referenced internal or instance-local Paperclip issues or links
- [x] My branch name describes the change and contains no internal Paperclip ticket ID or instance-derived details
- [x] I have run the affected tests locally and they pass
- [x] I have added or updated tests where applicable
- [x] I have documented the new contracts and compatibility boundary
- [x] I have considered and documented compatibility and security risks above
- [ ] All Paperclip CI gates are green
- [ ] Greptile is 5/5 with no open P2s, recommendations, or follow-ups
- [x] I will address all Greptile and reviewer comments before requesting merge

### CODE REVIEW

File: server/src/services/question-response-delivery.ts [hunk 1]
```diff
@@ -0,0 +1,685 @@
...
+      renewal = renewal.then(async () => {
+        if (stopped) return;
+        const renewedAt = now();
+        const renewed = await db.update(issueQuestionResponseDeliveries).set({
+          lastAttemptAt: renewedAt,
+          updatedAt: renewedAt,
+        }).where(and(
+          eq(issueQuestionResponseDeliveries.id, delivery.id),
+          eq(issueQuestionResponseDeliveries.status, "delivering"),
+        )).returning({ id: issueQuestionResponseDeliveries.id });
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply server/src/services/question-response-delivery.ts:423]: Fixed in 27d522202. `attemptCount` now acts as the fencing generation: lease renewal, retry release, and terminal writes all require the claimed generation, and side-effect completion revalidates ownership before the worker may continue. The new stale-worker regression reclaims attempt 1 as attempt 2 and verifies the expired worker cannot overwrite the newer receipt or emit a second delivery event.

File: server/src/services/question-response-delivery.ts [hunk 2]
```diff
@@ -0,0 +1,725 @@
...
+        if (!exhausted) return null;
+        return recordTerminal({
+          delivery: claimed,
+          interaction,
+          status: "failed",
+          mode: null,
+          targetRunId: null,
+          adapter,
+          errorCode,
+        });
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply server/src/services/question-response-delivery.ts]: Fixed in 0ded09fb1. A null wake now remains pending without consuming the bounded permanent-error budget; only actual wake exceptions can exhaust that budget. The regression holds suppression for five attempts, removes it on the sixth, and verifies the stored answer is then delivered with attemptCount 6.

**Comment by cryppadotta** [comment]
CI note: the 0ded09fb1 run checks out the current merge ref after #12270 changed `ui/package.json`, but master’s CI-owned `pnpm-lock.yaml` refresh has not landed yet. Every failing matrix job stops at the same frozen-lockfile install error before this PR’s tests run. The repository-owned refresh is #12310; this PR intentionally remains lockfile-free and will be rebased/rerun after that lands.

File: server/src/services/question-response-delivery.ts [hunk 3]
```diff
@@ -0,0 +1,723 @@
...
+    };
+  }
+
+  async function releaseForRetry(
+    delivery: DeliveryRow,
+    errorCode: string,
+    options: { bounded: boolean } = { bounded: true },
+  ) {
+    const at = now();
+    const exhausted = options.bounded && delivery.attemptCount >= MAX_DELIVERY_ATTEMPTS;
```
[the remark this answers is not in the document]
Comment (cryppadotta) [reply server/src/services/question-response-delivery.ts]: Fixed in ab37c1107. Claim generation (`attempt_count`) is now separate from bounded side-effect failures (`error_count`), so suppressed wakes do not consume the real-error retry budget. Added a regression covering five suppressed wakes, a transient error, and a subsequent successful delivery; focused suite is 26/26.

File: server/src/services/question-response-delivery.ts (hunk 1 above)
[the remark this answers is not in the document]
Comment (cryppadotta) [reply server/src/services/question-response-delivery.ts:423]: Follow-up in 03a1ef008 closes the remaining side-effect race. Question-answer wakes now use a new `question-response:` idempotency namespace protected by a partial unique DB index; a losing claimant recovers the winner’s durable wake instead of creating a second continuation or consuming an error retry. Added constraint and race regressions; focused suite is 28/28.

**Comment by cryppadotta** [comment]
Addressed the migration replay issue and completed the review loop.

- `issue_question_response_deliveries` now uses `CREATE TABLE IF NOT EXISTS`.
- Foreign-key creation is guarded with `duplicate_object` handling.
- All new indexes use `IF NOT EXISTS`.
- Added an embedded-Postgres regression test that replays the migration twice against an already migrated database.
- Also fixed Greptile's follow-up recovery edge case so a completed wake is reused after interrupted receipt finalization rather than retried or duplicated.

Verification on current head `7d2323045`: migration replay test passes; 29/29 focused delivery tests pass; repository typecheck and production build pass; all GitHub Actions/security checks pass; Greptile is 5/5. The PR remains lockfile- and workflow-free.
